src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx

```diff
@@ -113,18 +113,27 @@ function QueryEditor(props: QueryEditorProps) {
         LAST_USED_QUERY_ACTION_KEY,
     );
 
-    const [sendExecuteQuery, executeQueryResult] = executeQueryApi.useExecuteQueryMutation();
-    const [sendExplainQuery, explainQueryResult] = explainQueryApi.useExplainQueryMutation();
-    const [sendCancelQuery, cancelQueryResult] = cancelQueryApi.useCancelQueryMutation();
+    const [sendExecuteQuery] = executeQueryApi.useExecuteQueryMutation();
+    const [sendExplainQuery] = explainQueryApi.useExplainQueryMutation();
+    const [sendCancelQuery] = cancelQueryApi.useCancelQueryMutation();
```

Comment:
Let's not put cancel query status in the store

```suggestion
    const [sendCancelQuery, cancelQueryState] = cancelQueryApi.useCancelQueryMutation();
    
    const queryId = executeQuery.result?.queryId;
    const cancelStatus =
        queryId && cancelQueryState.originalArgs?.queryId === queryId
            ? cancelQueryState.status
            : undefined;

```

Reply:
Now `queryId` is in the state, so we can make a self contained `CancelButton` component.

Reply:
fixed